### PR TITLE
ignore SlotSets for requested slots

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed changes**:
- fix https://github.com/RasaHQ/rasa-sdk/issues/370
- fix https://github.com/RasaHQ/rasa/issues/7557
- the issue was that we included the `SlotSet("requested_slot", <value>)` when we checked whether the custom action extracted any slots

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by wochinge** [comment]
will cut a micro for this (once the 2.2.x branch is ready)

File: rasa/core/actions/forms.py [hunk 1]
```diff
@@ -455,7 +455,9 @@ async def validate(
         )
 
         some_slots_were_validated = any(
-            isinstance(event, SlotSet) for event in validation_events
+            isinstance(event, SlotSet)
+            for event in validation_events
+            if isinstance(event, SlotSet) and not event.key == REQUESTED_SLOT
```
Comment (Ghostvv) [inline rasa/core/actions/forms.py:460]: could you please add an explanatory comment.

**Review by Ghostvv — APPROVED** [review]
🚀

File: rasa/core/actions/forms.py (hunk 1 above)
Comment (wochinge) [reply rasa/core/actions/forms.py]: done

**Review by alwx — APPROVED** [review]
👍

**Pull request merged**